File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i=0; i<s.length(); i++){
            if(stack.isEmpty()){
                stack.push(s.charAt(i));
            }
            else if(s.charAt(i) ==')' && stack.peek()=='(' 
                   || s.charAt(i)=='}' && stack.peek()=='{'
                   || s.charAt(i)==']' && stack.peek()=='['){
                stack.pop();
            }
            else{
                stack.push(s.charAt(i));
            }
        }
        if(stack.isEmpty()){
            return true;
        }
        return false;
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        for(int i=0; i<s.length(); i++){
            if(stack.empty()){
                stack.push(s[i]);
            }
            else if(s[i]==')' && stack.top()=='(' 
                   || s[i]=='}' && stack.top()=='{'
                   || s[i]==']' && stack.top()=='['){
                stack.pop();
            }
            else{
                stack.push(s[i]);
            }
        }
        if(stack.empty()){
            return true;
        }
        return false;
    }
};